Improved the summary on the docs homepage (#1071)
cert-manager/website: content/docs/README.md

---
title: cert-manager
description: |
    cert-manager creates signed TLS / SSL certificates for workloads in your Kubernetes or OpenShift cluster
    and renews the certificates before they expire.
---

cert-manager creates signed TLS / SSL certificates for workloads in your Kubernetes or OpenShift cluster
and renews the certificates before they expire.

> Typically, the private key and signed certificate are stored in Kubernetes secrets

Reviewer: ```suggestion
The private key and signed certificate are stored in Kubernetes secrets and used by applications or ingress controllers. The csi-driver addon instead lets applications mount these certificates in its container directly, ideal for short-lived certificates.
```

and used by applications or ingress controllers.

> cert-manager can obtain signed certificates from a variety of certificate authorities, including:

Reviewer: ```suggestion
cert-manager can obtain certificates from a variety of certificate authorities, including:
```
Not sure that we have to mention that these certificates are signed, I think all certificates are signed.

> cert-manager can obtain the signed certificates from a variety of certificate authorities, including:

Reviewer 2: ```suggestion
cert-manager can obtain signed certificates from a variety of certificate authorities, including:
```

[Let's Encrypt](configuration/acme/), [HashiCorp Vault](configuration/vault/),

> [Venafi](configuration/venafi/) and [private PKI](configuration/ca/).

Author: When I previewed these links locally, they always ended up with a double forward slash between `docs//configuration`, no matter how I wrote the link. E.g.

http://localhost:8888/docs//configuration/acme/

Any ideas why nextjs is doing this?

Reviewer 3: https://github.com/cert-manager/website/#task-linking-to-other-documentation-pages

Does that help at all or is it something different to that?

Reviewer 2: does `./configuration/acme/` work?

Author: I have tried both suggestions, but neither of them seem to work.
I've pushed a version where I link to the `.md` files, but this still has the `docs//configuration` problem and additionally, the resulting URl for the Let's Encrypt link becomes `http://localhost:8888/docs//configuration/acme/README/`

Author: @PatrickHeneise Do you have any idea what is causing the URLs to do this? :point_up:

Reviewer 3: The style that I would expect to work would be `[Venafi](./configuration/venafi.md)` - everything linking to a markdown file with a relative link

Reviewer 3: It's not redundant in this scenario AIUI

It's possible that there's some bug in the main README which breaks links to other pages from there

Reviewer 4: Also locally, you can `Cmd+Click` on the link. If the file doesn't open, something is wrong.

Author: Thanks @PatrickHeneise , but is the case in this branch, and yet the URLs are wrong
 (double forward slashes and in the case of the acme link, a trailing README segment) 
See:
 * https://github.com/cert-manager/website/blob/baf88a31deb4c4dbd188e081d8e4fca65f47bf3b/content/docs/README.md
![image](https://user-images.githubusercontent.com/978965/189107499-e620ad71-83b8-4b0b-8a63-04c61f570076.png)

* https://deploy-preview-1071--cert-manager-website.netlify.app/docs/
![Screen Shot 2022-09-08 at 12 07 41](https://user-images.githubusercontent.com/978965/189107284-2c6ed781-e77c-44ca-b0d3-f08ddbc101a9.png)

![image](https://user-images.githubusercontent.com/978965/189107631-f1738f20-b5c7-4be4-831a-4c424c978637.png)

Author: @PatrickHeneise I suspect there's a bug in https://github.com/zentered/next-product-docs/blob/feat/cert-manager-adjustments/src/lib/remark-plugins/links/index.js , but I don't know how to make the nextjs / react / npm use a local copy of that package so that I can try and fix it.

Reviewer 4: Hm. Could you prefix with `./`?

I'll probably work on the lib next week and can fix that (and write docs on how to load it locally). In the meantime, please add your case here: https://github.com/zentered/next-product-docs/blob/feat/cert-manager-adjustments/test/remark-plugins/links.test.js

Those are all the various cases we collected over time and they produce the correct links.

Reviewer 4: We've started on the next release for next-product-docs :)

In the meantime, you can 
1. download the repo: `https://github.com/zentered/next-product-docs`
2. switch to the `feat/cert-manager-adjustments` branch
3. `pnpm install && pnpm build`
4. run `pnpm link .`
5. In the website repo, run `pnpm link @zentered/next-product-docs`

To boot up the development environment, run `pnpm watch` in the product-docs repo, that'll continuously build all the changes and make them available through the link.

Reviewer 4: There's a new and simplified version of the internal link resolver you can find here: https://github.com/zentered/next-product-docs/blob/main/src/lib/remark-plugins/links/index.js

Test cases: https://github.com/zentered/next-product-docs/blob/main/test/remark-plugins/links.test.js

Reviewer: see https://github.com/cert-manager/website/pull/1427

It can issue certificates from a variety of supported sources, including
[Let's Encrypt](https://letsencrypt.org), [HashiCorp Vault](https://www.vaultproject.io),
and [Venafi](https://www.venafi.com/) as well as private PKI.
cert-manager can obtain the signed certificates from a variety of certificate authorities, including:
[Let's Encrypt](configuration/acme/), [HashiCorp Vault](configuration/vault/),
[Venafi](configuration/venafi/) and [private PKI](configuration/ca/).

It will ensure certificates are valid and up to date, and attempt to
renew certificates at a configured time before expiry.

It is loosely based upon the work of
[kube-lego](https://github.com/jetstack/kube-lego) and has borrowed some
wisdom from other similar projects such as

> [kube-cert-manager](https://github.com/PalmStoneGames/kube-cert-manager).

Author: We decided to drop this. It doesn't seem important enough to include on the frontpage.

Reviewer 3: Agreed!

---
title: cert-manager
description: |
    cert-manager creates TLS certificates for workloads in your Kubernetes or OpenShift cluster and renews the certificates before they expire.
---

cert-manager creates TLS certificates for workloads in your Kubernetes or OpenShift cluster
and renews the certificates before they expire.

cert-manager can obtain certificates from a [variety of certificate authorities](configuration/issuers.md), including:

> [Let's Encrypt](configuration/acme/README.md), [HashiCorp Vault](configuration/vault.md),

Reviewer: ```suggestion
[Let's Encrypt](configuration/acme.md), [HashiCorp Vault](configuration/vault.md),
```
